src/System.Linq/src/System/Linq/Lookup.cs

```diff
@@ -247,8 +245,7 @@ public IEnumerable<TResult> ApplyResultSelector<TResult>(Func<TKey, IEnumerable<
                 do
                 {
                     g = g._next;
-                    g.Trim();
-                    yield return resultSelector(g._key, g._elements);
+                    yield return resultSelector(g._key, g);
```

Comment:
I think we should keep Trim here. There could be reasons why it was done. Most likely performance of iterating.
Also, while it is unfortunate to expose the array, there is a risk now that someone took dependency on that.